learning_flexbox/styles.scss

```diff
@@ -0,0 +1,198 @@
+* {
+    box-sizing: border-box;
+    padding: 0rem;
+    margin: 0rem;
+    height: 100%;
+
+
+}
+
+//Variables
+$baseblue:#04a3d3;
+$orange:#ff5722;
+$basetext:rgba(0, 0, 0, 0.8);
+// ______________FLEX AND NAV BAR__________
+
+a {
+    color: inherit;
+    text-decoration: none;
+}
+
+
+.flex-container {
+    position: absolute;
+    width: 100%;
+    display: flex;
+    padding-top: 1.7rem;
+    height: 5rem;
+    color: black;
+    font-size: 1.3rem;
+    background-color: rgba(206, 176, 65, 0.3);
+
+    a:hover,
+    a:focus {
+        border-bottom: 2px solid $basetext;
+        transition-duration: 0.5s;
+    }
+
+    .homepage {
+        margin-left: 2rem;
+        text-align: left;
+    }
+
+    .collection {
+        margin-right: auto;
+        margin-left: 2rem;
+        text-align: left;
+    }
+
+    .search {
+        text-align: right;
+        margin-right: 1rem;
+    }
+
+    .account {
+        margin-left: auto;
+        margin-right: 2rem;
+        text-align: right;
+    }
+
+    .cart {
+        margin-right: 2em;
+        text-align: right;
+    }
+
+    .bag {
+        margin-right: 2rem;
+        text-align: right;
+
+    }
+}
+
+// .container>div {
+//     flex: 1;
+//     /*flex-grow, flex-shrink, flex-basts*/
+// }
+
+.flex-container>.collection {
+    flex: 2;
+    /*flex-grow, flex-shrink, flex-basis*/
+}
+
+.flex-container>.account {
+    flex: 2;
+}
+
+
+
+//______________GRID AND LAYOUT_____________
+
+.grid-container {
+    display: grid;
+    grid-template-columns: repeat(15, 1fr);
+    grid-template-rows: repeat(10, 1fr);
+
+    .media {
+        background-color: $orange;
+        grid-column: 1/2;
+        grid-row: 1/-1;
+        display: flex;
+        flex-direction: column;
+        text-align: center;
+        padding-top: 7rem;
+
+        .social {
```

Comment:
Here's the bad thing about SCSS: Too much Nesting.

It's so easy to nest classes that we forget its implications. Only do nesting if it's really necessary, otherwise keep it flat.

```
// bad: class nesting
.grid-container .media

// better: only the child class
.media
// or creating a class based on its context (parent + element).
.grid-container-media 
```

When writing a lot of CSS, its recommended to follow some "CSS methodology" to keep CSS organized.
These methodologies are a set of naming conventions to follow when creating CSS classes.

[BEM](http://getbem.com/introduction/) is the CSS methodology most known, although [SUITCSS](https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md) is my favorite. Add it to your learnings and apply it on a next challenge :)

It doesn't matter much what methodology you choose, as long as you follow it through the project.